dronekit/lib/__init__.py

```diff
@@ -273,28 +267,28 @@ def __init__(self):
     """
     Provides callback based notification on attribute changes.
 
-    The argument list for observer is ``observer(attr_name)``.
+    The argument list for observer is ``observer(object, attr_name)``.
     """
-    def add_attribute_observer(self, attr_name, observer):
+    def on_attribute(self, attr_name, observer):
         """
-        Add an attribute observer.
+        Add an attribute listener.
 
-        The observer callback function is called with the ``attr_name`` argument. 
+        The listener callback function is called with the ``vehicle`` and ``attr_name`` arguments.
         This can be used to infer the related attribute if the same callback is used 
         for watching several attributes.
-        
+
         The example below shows how to get callbacks for location changes:
 
         .. code:: python
 
             #Callback to print the location in global and local frames
-            def location_callback(location):
+            def location_callback(vehicle, location):
```

Comment:
in this case this should be `def location_callback(vehicle, attr_name):` see `attr_name` instead of `location`, `attr_name` is a string